### ISSUE TITLE
Unable to paste images in the latest version of Sanity UI

### ISSUE THREAD

**Issue by stephanepericat** [sanity-io/sanity#7161]
If you find a security vulnerability, do NOT open an issue. Email [security@sanity.io](mailto:security@sanity.io) instead.

**Describe the bug**

In version 2.6.8 of Sanity UI, I run into an issue where I am no longer able to paste images onto image fields.
First, Chrome / Brave asks me for additional permissions:

<img width="357" alt="Screenshot 2024-07-15 at 6 24 10 PM" src="https://github.com/user-attachments/assets/1855fbb5-1f8c-4eeb-a54d-e28a783d656c">

After allowing it, I get a toast notification that I have nothing to paste:

<img width="827" alt="Screenshot 2024-07-15 at 5 54 02 PM" src="https://github.com/user-attachments/assets/07e5a9d0-2a58-4e09-9484-99113129d339">

These are the package.json dependencies:

```
"@sanity/google-maps-input": "^4.0.1",
"@sanity/icons": "^3.3.0",
"@sanity/ui": "^2.6.8",
"@sanity/vision": "^3.50.0",
"react": "^18.3.1",
"react-dom": "^18.3.1",
"react-is": "^18.3.1",
"sanity": "^3.50.0",
"sanity-plugin-internationalized-array": "^2.0.0",
"sanity-plugin-media": "^2.2.5",
"styled-components": "^6.1.11"
```

It works fine if i revert to version 2.1.4 of `@sanity/ui` and 3.38.0 of `sanity`

**To Reproduce**

1. Create a project with latest version of Sanity packages
2. Create a document type that contains an image field
3. Run the UI locally, open in Brave, Chrome or Safari
4. try to paste an image into the image field

**Expected behavior**

I should be able to paste an image into an image field, as in previous versions.

**Screenshots**
See above.

**Which versions of Sanity are you using?**

```
@sanity/cli (global)          3.50.0 (up to date)
@sanity/eslint-config-studio   4.0.0 (up to date)
@sanity/google-maps-input      4.0.1 (up to date)
@sanity/icons                  3.3.0 (up to date)
@sanity/ui                     2.6.8 (up to date)
@sanity/vision                3.50.0 (up to date)
sanity                        3.50.0 (up to date)
```

**What operating system are you using?**

MacOS (happens both locally and when hosted on Vercel, including for Windows users)

**Which versions of Node.js / npm are you running?**

9.8.1
v18.18.0

**Additional context**

n/a

**Security issue?**

Any security issues should be submitted directly to [security@sanity.io](mailto:security@sanity.io). In order to determine whether you are dealing with a security issue, ask yourself these two questions:

- Can I access something that's not mine, or something I shouldn't have access to?
- Can I disable something for other people? If the answer to either of those two questions are "yes", then you're probably dealing with a security issue. Note that even if you answer "no" to both questions, you may still be dealing with a security issue, so if you're unsure, just email us at [security@sanity.io](mailto:security@sanity.io.

**Comment by cngonzalez** [comment]
Thanks for reporting @stephanepericat ! I've been able to reproduce this issue and believe the bug was introduced in Sanity version 3.50.0. We will triage this internally and you'll be alerted in this thread when this is resolved. Thanks again!

**Comment by stephanepericat** [comment]
@cngonzalez any updates?

**Comment by cngonzalez** [comment]
Ah, apologies @stephanepericat  -- we addressed this a few weeks ago (when our Linear bot closed the issue) but we should have closed this with a more informative comment, apologies for that!

 Yes, the issue should have been resolved in version [3.52.4](https://github.com/sanity-io/sanity/releases/tag/v3.52.4) -- so if you upgrade to any version after that, it should be resolved.

Are you seeing this behavior still after updating to a version after 3.52.4?
